### DISCUSSION TITLE
Are memory pools sorted?

### DISCUSSION THREAD

**Discussion by gpalmer-latai** [eclipse-iceoryx/iceoryx#2086] [Q&A] [answered]
This is more of a sanity check than anything but, I'm looking at [MemoryManager::getChunk](https://github.com/eclipse-iceoryx/iceoryx/blob/443a0d855e0592f6f19849527f6de98672b8089f/iceoryx_posh/source/mepoo/memory_manager.cpp#L154-L164) right now and to me it looks like the first memory pool is selected which has a chunk big enough to accomodate a message. 

Does this mean that if my roudi config has

```
[[segment]]

[[segment.mempool]]
size = 100000
count = 10

[[segment.mempool]]
size = 512
count = 1000
```

And i loan a small message, it'll select a chunk from the first memory pool to publish into?

Or am I missing some sort of sorting logic somewhere?

**Comment by elBoberido** [comment, marked as answer]
I should be sorted, else `MemoryManager::addMemPool` would abort with an error. It is done in `MePooConfig::optimize`.

**Reply by gpalmer-latai** [reply]
Makes sense, I see the logic now. Thanks!
